osfoffline/gui/qt/preferences.py

```diff
@@ -148,13 +149,14 @@ def selector(self, selected_index):
 
             self._executor = QtCore.QThread()
             self.node_fetcher = NodeFetcher()
-            self.treeWidget.setCursor(QtCore.Qt.BusyCursor)
+            QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
```

Comment:
Interesting. It may depend on what function you tried to decorate. (QT signals and slots are kind of their own world under the hood, for example)

Also, you may want to move the `logger.exception` call above the `raise e` statement so it actually fires. :)